### ISSUE TITLE
test_subprocess_send_signal() of test_asyncio hangs if SIGHUP is ignored

### ISSUE THREAD

**Issue by vstinner** [python/cpython#75217]
BPO | [31034](https://bugs.python.org/issue31034)
--- | :---
Nosy | @vstinner, @1st1, @stratakis
PRs | <li>python/cpython#2867</li><li>python/cpython#2876</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-07-26.00:35:00.539>
created_at = <Date 2017-07-25.15:11:10.614>
labels = ['3.7', 'tests', 'expert-asyncio']
title = 'test_subprocess_send_signal() of test_asyncio hangs if SIGHUP is ignored'
updated_at = <Date 2017-07-26.00:35:00.520>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-07-26.00:35:00.520>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-07-26.00:35:00.539>
closer = 'vstinner'
components = ['Tests', 'asyncio']
creation = <Date 2017-07-25.15:11:10.614>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 31034
keywords = []
message_count = 5.0
messages = ['299094', '299120', '299172', '299178', '299179']
nosy_count = 3.0
nosy_names = ['vstinner', 'yselivanov', 'cstratak']
pr_nums = ['2867', '2876']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31034'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
A colleague reported me that the Python test suite hangs on running test_subprocess_send_signal() of test_asyncio. After analysing the issue, it seems like the test hangs becaues the RPM package builder ignores SIGHUP using something like:


```py
def daemonize():
    [...]
    signal.signal(signal.SIGHUP, signal.SIG_IGN)
    [...]
```


While we should see if Koji (software used to build packages) can be patched to restore the default signal handler for SIGHUP, I would like to enhance test_asyncio to not depend on the current SIGHUP signal handler.

I modified [Lib/site.py](https://github.com/python/cpython/blob/main/Lib/site.py) to add "import signal; signal.signal(signal.SIGHUP, signal.SIG_IGN)": on 406 test files, only test_asyncio hangs.

Attached PR fixes test_asyncio.

**Comment by vstinner** [comment]
New changeset 830080913c22a9834d310294b9f7653234dc6a59 by Victor Stinner in branch 'master':
bpo-31034: Reliable signal handler for test_asyncio (bpo-2867)
https://github.com/python/cpython/commit/830080913c22a9834d310294b9f7653234dc6a59

**Comment by vstinner** [comment]
Patch to test manually the fix:

```diff
diff --git a/Lib/test/test_asyncio/__init__.py b/Lib/test/test_asyncio/__init__.py
index 80a9eea..1418952 100644
--- a/Lib/test/test_asyncio/__init__.py
+++ b/Lib/test/test_asyncio/__init__.py
@@ -1,3 +1,4 @@
+import signal; signal.signal(signal.SIGHUP, signal.SIG_IGN)
 import os
 from test.support import load_package_tests, import_module
```

**Comment by vstinner** [comment]
New changeset efe9fcbd2ca85ba9d6af6d95cc530a9c332f37c5 by Victor Stinner in branch '3.6':
bpo-31034: Reliable signal handler for test_asyncio (bpo-2867) (bpo-2876)
https://github.com/python/cpython/commit/efe9fcbd2ca85ba9d6af6d95cc530a9c332f37c5

**Comment by vstinner** [comment]
Ok, I fixed Python 3.6 and master (future 3.7). Python 3.5 is slowly entering security-only fixes. I don't think that this corner case is improtant enough to justify a backport.
